src/o365/graph/commands/teams/teams-messagingsettings-set.ts

```diff
@@ -0,0 +1,176 @@
+    });
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        const body: any = {
+          messagingSettings: {}
+        };
+        GraphTeamsMessageSettingsSetCommand.props.forEach(p => {
+          if (typeof (args.options as any)[p] !== 'undefined') {
+            body.messagingSettings[p] = (args.options as any)[p] === 'true';
+          }
+        });
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          }),
+          body: body,
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.patch(requestOptions);
+      })
+      .then((): void => {
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The ID of the Microsoft Teams team for which to update message settings'
+      },
+      {
+        option: '--allowUserEditMessages  [allowUserEditMessages ]',
+        description: 'Set to true to allow users to edit messages and to false to disallow it'
+      },
+      {
+        option: '--allowUserDeleteMessages [allowUserDeleteMessages]',
+        description: 'Set to true to allow users to delete messages and to false to disallow it'
+      },
+      {
+        option: '--allowOwnerDeleteMessages [allowOwnerDeleteMessages]',
+        description: 'Set to true to allow owner to delete messages and to false to disallow it'
+      },
+      {
+        option: '--allowTeamMentions [allowTeamMentions]',
+        description: 'Set to true to allow @team or @[team name] mentions and to false to disallow it'
+      },
+      {
+        option: '--allowChannelMentions [allowChannelMentions]',
+        description: 'Set to true to allow @channel or @[channel name] mentions and to false to disallow it'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.teamId) {
+        return 'Required parameter teamId missing';
+      }
+
+      if (!Utils.isValidGuid(args.options.teamId)) {
+        return `${args.options.teamId} is not a valid GUID`;
+      }
+
```

Comment:
Does it occur only in commands that allow unknown options @VelinGeorgiev?

Reply:
Not really. Consider that scenario

`o365 graph teams messagingsettings set --teamId 'abc1' --teamId 'abc2'`

User can accidently repeat an option and It is a valid scenario for `vropal`, but instread of teamId='abc2', it returns teamId=['abc1','abc2'] and we usually do not handle that in all our commands.

Reply:
I get your point. I see it as an edge case, especially since you're specifying the options yourself, but there is no harm in handling it centrally as you mentioned. Let's add a separate issue for it to track it there.